content/blog/2023/2023-05-03-react-and-spring-boot.md

```diff
@@ -0,0 +1,704 @@
+
+Axios is a popular JavaScript library that provides an easy-to-use interface for making HTTP requests from the client side. It is highly configurable, and it supports various request methods, such as GET, POST, PUT, DELETE, and more. With Axios, we can easily fetch data from external APIs and update our application's state accordingly.
+
+Additionally, we will install Bootstrap, a popular CSS and JavaScript library that provides a collection of components, such as buttons, forms, modals, and more, to easily style our React components.
+
+Paste the following code in the React application's terminal:
+
+```bash
+npm install axios bootstrap
+```
+
+The above code will install the Axios and Bootstrap dependency into our application. All dependency files are saved in the **node\_modules** folder.
+
+To structure our React project, Run the following command to create the necessary folders and files for the application.
+
+```bash
+mkdir src/api src/components
+touch src/api/axiosConfig.js src/components/PublisherCrud.jsx src/components/PublisherList.jsx
+```
+
+Our React application structure will now look like this:
+
+{{% image alt="reactjs-project-structure" src="images/posts/spring-boot-and-reactjs/react.png" %}}
+
+The `src` folder, is where we will write our code. With the command above we have created components file `PublisherCrud.jsx` and `PublisherList.jsx`.
+
+JSX is a syntax extension to JavaScript, used by ReactJS for creating user interfaces. It allows developers to write HTML-like syntax directly in their JavaScript code, making it easier to visualize and manipulate the UI elements.
+
+Our App.js and App.css are components of our React application. App.js is the default landing page where we can define, pass and render all our application components UI.
+
+Noticed, we've created an `api` folder inside the `src` directory, containing an `axiosConfig.js` file. This is where our application will make all API calls.
+
+Next, In the `axiosConfig.js` file we will create a connection to our SpringBoot endpoints.
+
+To achieve this, Paste the following code in the `axiosConfig.js` file:
+
+```javascript
+import  axios  from  'axios';  
+
+export default axios.create({
+    baseURL: "http://localhost:8001/publisher",
+});
+```
+
+In the above code, we are using Axios dependency's `.create()` method.
+
+This method allows us to set default values for headers, timeouts, interceptors, and other properties that will be applied to all requests made by that instance. It is useful when we have to make multiple requests to the same API or when we need to customize the request behavior for a particular endpoint.
+
+With this, we can make calls to our Spring Boot endpoints without having to explicitly specify our full backend URL everywhere in our frontend code.
```

Comment:
please refer this article for more infromation https://reflectoring.io/tutorial-guide-axios/